cmd/truchaind/migrate.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"time"
+
+	"github.com/TruStory/truchain/cmd/truchaind/migration/v0_3"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	extypes "github.com/cosmos/cosmos-sdk/x/genutil"
+)
+
+const (
+	flagGenesisTime     = "genesis-time"
+	flagChainID         = "chain-id"
+	flagResetValidators = "reset-validators"
+)
+
+var migrationMap = extypes.MigrationMap{
+	"v0.3": v0_3.Migrate,
+}
+
+// GetMigrationCallback returns a MigrationCallback for a given version.
+func GetMigrationCallback(version string) extypes.MigrationCallback {
+	return migrationMap[version]
+}
+
+// GetMigrationVersions get all migration version in a sorted slice.
+func GetMigrationVersions() []string {
+	versions := make([]string, len(migrationMap))
+
+	var i int
+	for version := range migrationMap {
+		versions[i] = version
+		i++
+	}
+
+	sort.Strings(versions)
+	return versions
+}
+
+// MigrateGenesisCmd returns a command to execute genesis state migration.
+// nolint: funlen
+func MigrateGenesisCmd(_ *server.Context, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tru_migrate [target-version] [genesis-file]",
+		Short: "Migrate genesis to a specified target version",
+		Long: fmt.Sprintf(`Migrate the source genesis into the target version and print to STDOUT.
+Example:
+$ %s migrate v0.36 /path/to/genesis.json --chain-id=cosmoshub-3 --genesis-time=2019-04-22T17:00:00Z
```

Comment:
Change to `$ %s tru_migrate v0.3.1 /path/to/genesis.json --chain-id=betanet-2 --genesis-time=2019-04-22T17:00:00Z`